<launch>
    <arg name="veh" default="robot"/> 
    <arg name="radius" default="0.5"/>
    <arg name="lidar_topic" default="points"/>
    <arg name="joy" default="joy_teleop/joy"/>
    <arg name="cmd_vel" default="joy_teleop/cmd_vel"/>

    <group ns="$(arg veh)">
        <node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan">
            <remap from="cloud_in" to="$(arg lidar_topic)"/>
            <remap from="scan" to="RL/scan" />
            
            <rosparam>
                target_frame: front_laser       <!-- Leave disabled to output scan in pointcloud frame -->
                transform_tolerance: 0.01
                min_height: -0.3
                max_height: 0.5

                angle_min: -1.5708      <!-- -M_PI/2-->
                angle_max: 1.5708       <!-- M_PI/2-->
                angle_increment: 0.00437  <!-- M_PI/360.0 (every degrees)-->
                time_increment: 0.0
                scan_time: 0.0
                range_min: 0.10000000149
                range_max: 20.0
                use_inf: true
            </rosparam>
        </node>

        <node pkg="navigation" type="waypoint_navigation_obstacle_husky.py" name="waypoint_navigation_obstacle" respawn="false" output="screen">
            <param name="robot_radius" value="$(arg radius)" />
            <param name="joy" value="$(arg joy)" />
            <param name="cmd_vel" value="$(arg cmd_vel)" />
            <param name="veh" value="$(arg veh)" />
        </node>
    </group>

</launch>